lightning/src/ln/interactivetxs.rs

```diff
@@ -124,46 +124,50 @@ impl AbortReason {
-			},
-			AbortReason::InvalidTx => "The transaction is invalid".into(),
-			AbortReason::MissingFundingOutput => "No shared funding output found".into(),
-			AbortReason::DuplicateFundingOutput => "More than one funding output found".into(),
-			AbortReason::InvalidLowFundingOutputValue => {
-				"Local part of funding output value is greater than the funding output value".into()
-			},
-			AbortReason::ProvidedInputsAndPrevtxsTxIdMismatch(idx) => format!(
+		match self {
+			AbortReason::ProvidedInputsAndPrevtxsTxIdMismatch(idx) => f.write_fmt(format_args!(
 				"TxId mismatch in the provided inputs and previous transactions, input index {}",
 				idx
-			),
-			AbortReason::ProvidedInputsAndPrevtxsVoutNotFound(idx) => format!(
+			)),
+			AbortReason::ProvidedInputsAndPrevtxsVoutNotFound(idx) => f.write_fmt(format_args!(
 				"Vout provided in an input is not found in the previous transaction, input index {}",
-				idx),
-			AbortReason::InternalErrorGettingDestinationScript => {
-				"Internal error getting destination script".into()
-			},
-		})
+				idx
+			)),
+			_ => f.write_str(&match self {
+				AbortReason::InvalidStateTransition => "State transition was invalid",
+				AbortReason::UnexpectedCounterpartyMessage => "Unexpected message",
+				AbortReason::ReceivedTooManyTxAddInputs => "Too many `tx_add_input`s received",
+				AbortReason::ReceivedTooManyTxAddOutputs => "Too many `tx_add_output`s received",
+				AbortReason::IncorrectInputSequenceValue => {
+					"Input has a sequence value greater than 0xFFFFFFFD".into()
+				},
+				AbortReason::IncorrectSerialIdParity => "Parity for `serial_id` was incorrect",
+				AbortReason::SerialIdUnknown => "The `serial_id` is unknown",
+				AbortReason::DuplicateSerialId => "The `serial_id` already exists",
+				AbortReason::PrevTxOutInvalid => "Invalid previous transaction output",
+				AbortReason::ExceededMaximumSatsAllowed => {
+					"Output amount exceeded total bitcoin supply".into()
+				},
+				AbortReason::ExceededNumberOfInputsOrOutputs => "Too many inputs or outputs",
+				AbortReason::TransactionTooLarge => "Transaction weight is too large",
+				AbortReason::BelowDustLimit => "Output amount is below the dust limit",
+				AbortReason::InvalidOutputScript => "The output script is non-standard",
+				AbortReason::InsufficientFees => "Insufficient fees paid",
+				AbortReason::OutputsValueExceedsInputsValue => {
+					"Total value of outputs exceeds total value of inputs".into()
+				},
+				AbortReason::InvalidTx => "The transaction is invalid",
+				AbortReason::MissingFundingOutput => "No shared funding output found",
+				AbortReason::DuplicateFundingOutput => "More than one funding output found",
+				AbortReason::InvalidLowFundingOutputValue => {
+					"Local part of funding output value is greater than the funding output value"
+				},
+				AbortReason::InternalErrorGettingDestinationScript => {
+					"Internal error getting destination script"
+				},
+				_ => unreachable!(),
```

Comment:
Were somewhat adverse to using `unreachable` especially when it may later become reachable. Let's just call `write_str` for each variant and combine the `match`es into one.

Reply:
OK, simplified